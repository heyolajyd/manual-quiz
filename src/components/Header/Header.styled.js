import styled from 'styled-components';

export const HeaderContainer = styled.div``;

export const HeaderTitle = styled.h1`
  color: ${(props) => props.theme.colors.grey};
`;
