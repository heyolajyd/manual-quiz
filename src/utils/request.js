/**
 * Util function to get questions
 */
export const getQuestion = async () => {
  try {
    const res = await fetch('./questions.json');
    return await res.json();
  } catch (err) {
    console.log('Failed to parse: ', JSON.stringify(err));
    throw new Error(err);
  }
};
