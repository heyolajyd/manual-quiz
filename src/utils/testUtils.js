import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
