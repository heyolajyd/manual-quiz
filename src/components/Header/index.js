import React from 'react';
import PropTypes from 'prop-types';
import { HeaderContainer, HeaderTitle } from './Header.styled';

const Header = ({ questionNumber, questionCount }) => {
  return (
    <HeaderContainer>
      <HeaderTitle>
        Question {questionNumber + 1} of {questionCount}
      </HeaderTitle>
    </HeaderContainer>
  );
};

Header.propTypes = {
  questionNumber: PropTypes.number,
  questionCount: PropTypes.number,
};

export default Header;
