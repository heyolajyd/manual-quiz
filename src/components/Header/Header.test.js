import React from 'react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import Header from './index';

describe('Header', function () {
  beforeEach(() => {
    const { container, getByText } = withThemeProvider(
      <Header questionCount={4} questionNumber={0} />
    );

    this.container = container;
    this.getByText = getByText;
  });

  it('should match snapshot', () => {
    expect(this.container).toMatchSnapshot();
  });

  it('should display question number and question count', () => {
    expect(this.getByText(/Question 1 of 4/)).toBeInTheDocument();
  });
});
