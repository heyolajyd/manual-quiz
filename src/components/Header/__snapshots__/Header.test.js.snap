// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
.c0 {
  color: #494949;
}

<div>
  <div
    class=""
  >
    <h1
      class="c0"
    >
      Question 
      1
       of 
      4
    </h1>
  </div>
</div>
`;
